### ISSUE TITLE
Migrate website away from wiki.gnome.org

### ISSUE THREAD

**Issue by SqAtx** [getting-things-gnome/gtg#1092]
The website is at https://wiki.gnome.org/Apps/GTG, which now has a banner that the site is planned for retirement:

![image](https://github.com/getting-things-gnome/gtg/assets/4351932/a8f24d89-0ce1-4b7c-bd3d-c0bb60a0cd2e)

The [migration guide](https://hedgedoc.gnome.org/wiki-migration-guide#) recommends migrating to https://apps.gnome.org

**Comment by nekohayo** [comment]
From a marketing standpoint, what I want to avoid is loss of control due to old content lingering on the read-only archived version of the GNOME Wiki eventually as it would become carbon-frozen and non-updateable; I would want our migration to delete all the old contents and put redirects in place.

**Comment by diegogangl** [comment]
The best solution considering visiblity would be apps.gnome.org but I'm guessing you have to be accepted.

There's also Github pages but I think the theme selection is more limited. If we go with the website way, I would make a very simple one page that links to flathub page, and the readme files in the docs for contributor stuff

**Comment by Neui** [comment]
GitHub pages is basically just HTML files, we can have any theme. Jekyll integration is just an integration, not an requirement.

**Comment by picsel2** [comment]
Deadline incoming 😱: https://discourse.gnome.org/t/wiki-gnome-org-will-be-retired-on-23-october-2024/24543

**Comment by Neui** [comment]
I've made a (supposed to be quick) proof-of-concept: https://neui.github.io/gtg/ (from https://github.com/Neui/gtg/tree/gh-pages)

We could create a repository [named `getting-things-gnome.github.io`](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site#creating-a-repository-for-your-site) to use [gettings-things-gnome.github.io](https://gettings-things-gnome.github.io) as a domain rather than in a subdirectory (since my repo is named gtg, it is in the `gtg` subdirectory on "my" domain).
It is using GitHubs jekyll integration mainly because that is what they advertise and I wanted to use Markdown (I think it would be enough for us). It appears we can use a different generator with Github Actions (or GitLab CI if we swtich over).
I used [pandoc](https://pandoc.org/) to convert the HTML from the wiki to Markdown and cleaned it up manually.

**Comment by Neui** [comment]
I don't think the old wiki content should be "deleted" but rather let there be a banner (on important pages?) to the new URL. So that old links still work, you can maybe read the contents if the new site isn't available, and possible conversion mistakes on my side.

I can't do a PR here because the `gh_pages` branch doesn't exists. (Unless you want me to make a `docs` subdirectory under the `master` branch, which the [pages settings](https://github.com/getting-things-gnome/gtg/settings/pages) suggests is possible).

There is also the question if we should put the wiki directly under https://getting-things-gnome.github.io (which means a [new repository](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site#creating-a-repository-for-your-site) called `getting-things-gnome/getting-things-gnome.github.io` should be created which contains the site) or in a subdirectory like I have (https://getting-things-gnome.github.io/gtg/, and pushing into the `gh_pages` branch of this repository).
I would do the extra repository one so it's separate from the main repository and has a cleaner url (no subdirectory, the domain name already implicitly contains gtg).

**Comment by nekohayo** [comment]
> I don't think the old wiki content should be "deleted" but rather let there be a banner (on important pages?) to the new URL. So that old links still work, you can maybe read the contents if the new site isn't available, and possible conversion mistakes on my side.

Indeed, the pages wouldn't be completely deleted from the wiki (i.e. not create 404 there), but instead I would replace their contents entirely by a short message / URL pointing to the new website here on github.

As you mentioned elsewhere, GitHub pages assigns the subdomain based on the username/org name, and we don't get to pick a custom one. In any case, I think we will make do with this, and worst case scenario if we are ever asked/forced to change the name someday we can probably do that on our side… the link that the GNOME Wiki would link to would break, but then that would be _their_ problem for not having updatable things anymore, at least we wouldn't have a bunch of immutable contents in there ;)

> I would do the extra repository one so it's separate from the main repository and has a cleaner url (no subdirectory, the domain name already implicitly contains gtg).

I agree that it should probably be a separate repository rather than within the code repository, to keep things cleaner (and it allows converting to some other website at any other time anyway). Let's try to do that and see if it works.

**Comment by nekohayo** [comment]
Thanks to @Neui's fantastic conversion/preservation work, all the contents have been successfully moved to https://getting-things-gnome.github.io, then I went through each page under `https://wiki.gnome.org/Apps/GTG/` to modify their contents to point to the new location (with also a 5 to 7 seconds redirect). I edited 58 pages that way, hopefully I didn't forget any 🤞

Some pages redirect to the precise replacement URL (like the handful of core pages that were made post-2020, and the liblarch reference pages), some point to a topic index (like the GSoC ones), and most point just to the new home page (especially for very old contents), because I have no guarantee that links will always keep the same location and structure in the future.

There are a few issues remaining, listed in https://github.com/getting-things-gnome/getting-things-gnome.github.io/issues, but I think we should be good for now, overall.

It would also probably make sense for @diegogangl to remove https://github.com/getting-things-gnome/gtg/wiki if that folder in the main code repository was meant as a stopgap solution. With that (or at least a decision on that front), I think we could close this ticket.

**Comment by diegogangl** [comment]
Removed the wiki from the repo 👍
